hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveConf.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.hive;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * For https://github.com/apache/iceberg/issues/2070
+ */
+public class TestHiveConf {
+  private static final String HIVE_SITE_CONTENT = "<?xml version=\"1.0\"?>\n" +
+      "<?xml-stylesheet type=\"text/xsl\" href=\"configuration.xsl\"?>\n" +
+      "<configuration>\n" +
+      "  <property>\n" +
+      "    <name>hive.metastore.sasl.enabled</name>\n" +
+      "    <value>true</value>\n" +
+      "  </property>\n" +
+      "</configuration>\n";
+
+  @Test
+  public void test() {
+    HiveConf hiveConf = createHiveConf();
```

Comment:
Wouldn't it be better to have a test for the HiveClientPool specifically? We want to test that the clients in the pool has the specific configuration we need. What we doing here is testing that the HiveConf works as we expect. With some obscure change we can end up in a situation where the HiveConf behaviour does not change but the HMS Clients do not het the conf we need.
Thanks, Peter

Reply:
I added TestHiveClientPool, please take a look when you have time.

Reply:
@pvary the CI failed occasionally for org.apache.thrift.transport.TTransportException in spark-catalog, for example：

![image](https://user-images.githubusercontent.com/1303530/104606144-5bdc4480-56ba-11eb-9bf3-f7329137c2ec.png)

I think this may thrown by a hive metastore client, and maybe related to the code in org.apache.iceberg.hive.TestHiveMetastore#start(int), what do you think?
```java
// in Hive3, setting this as a system prop ensures that it will be picked up whenever a new HiveConf is created
System.setProperty(HiveConf.ConfVars.METASTOREURIS.varname, hiveConf.getVar(HiveConf.ConfVars.METASTOREURIS));
```
BTW, both Hive2 and Hive3 will pick up system property whenever a new HiveConf is created as i tested.

Reply:
There are some issues with how we handle pools in general. There is some work fixing this on another PR

Reply:
> There are some issues with how we handle pools in general. There is some work fixing this on another PR

Glad to hear that!